### PR TITLE
allow passing of GET options to .all in collections

### PR DESCRIPTION
**Pull request by durandom**
adds ability to pass down options to collections

```ruby
filter = {
 :order_by      => 'timestamp',
 :timestamp__gt => '2016-08-02T17:56:37.212874Z'
}
connection.api.activity_stream.all(filter)
```

see sorting, searching and filtering in the tower api docs https://docs.ansible.com/ansible-tower/3.0.1/html/towerapi

@miq-bot assign @syncrou

### CODE REVIEW

**Comment by bdunne** [comment]
@durandom Can you update the description with some examples of get_options?

**Comment by durandom** [comment]
> @durandom Can you update the description with some examples of get_options?

sure, sorry for doing this in the first place...
I also added a bit of yardoc to the method, not sure if thats wanted though

**Comment by syncrou** [comment]
👍 LGTM

File: lib/ansible_tower_client/collection.rb [hunk 1]
```diff
@@ -6,17 +6,20 @@ def initialize(api, klass = nil)
       @klass = klass
     end
 
-    def all
-      find_all_by_url(klass.endpoint)
+    # @param get_options [Hash] a hash of http GET params to pass to the api request
+    #   e.g. { :order_by => 'timestamp', :name__contains => 'foo' }
+    def all(get_options = nil)
```
Comment (syncrou) [inline lib/ansible_tower_client/collection.rb]: @durandom - I had some concerns about passing args to the `all` method - but it seems this is the same behavior as ActiveRecord.  So I'm fine with it.

**Pull request merged**